File: HeadFirst/HeadFirst/ProxyPattern/GumballMonitor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HeadFirst.ProxyPattern
{
    public class GumballMonitor
    {
        private GumballMachine machine;

        public GumballMonitor(GumballMachine machine)
        {
            this.machine = machine;
        }

        public void Report()
        {
            Console.WriteLine("Gumball machine: " + machine.location);
        }
    }
}

File: DPJava/src/com/company/ProxyPattern/GumballMonitor.java
package com.company.ProxyPattern;

import java.rmi.RemoteException;

public class GumballMonitor {
    GumballMachineRemote machine;

    public GumballMonitor(GumballMachineRemote machine) {
        this.machine = machine;
    }

    public void report() {
        try {
            System.out.println("Gumball machine: " + machine.getLoacion());
        } catch (RemoteException e) {
            e.printStackTrace();
        }
    }
}
